### PR TITLE
ref: separate slurm from ddp

### PR DESCRIPTION
**Pull request by williamFalcon**
ref: separate slurm from ddp

### CODE REVIEW

File: pytorch_lightning/accelerators/ddp_slurm_backend.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    def early_stopping_should_stop(self, pl_module):
+        stop = torch.tensor(int(self.trainer.should_stop), device=pl_module.device)
+        dist.all_reduce(stop, op=dist.reduce_op.SUM)
+        dist.barrier()
+        should_stop = stop == self.trainer.world_size
+        return should_stop
+
+    def broadcast(self, obj, src=0):
+        return self.dist.broadcast(obj)
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/ddp_slurm_backend.py:99]: why's src unused?

Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_slurm_backend.py:99]: this is an override of the parent method in Accelerator. It's to keep behavior consistent
across accelerators.

src is already set into this object automatically.

File: pytorch_lightning/accelerators/ddp_slurm_backend.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+else:
+    HYDRA_AVAILABLE = True
+
+
+# -------------------------------------------
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# TEMP CLASS WHILE WE DECOUPLE SLURM FROM DDP
+# !!!!!!!!!!!!!! NOTE !!!!!!!!!!!!!!!!!!!!!!
+# -------------------------------------------
+class DDPSLURMBackend(Accelerator):
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/ddp_slurm_backend.py:41]: what's the final envisioned setup for slurm + DDP?

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/accelerators/ddp_slurm_backend.py (hunk 2 above)
Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_slurm_backend.py:41]: te is already split out :)

I think right now it's to finish fixing DDP and having 3 types of ddp tied to the same file is making debugging impossible.

Once those tests pass, i'll fix the TE cpu ddp.

Then we'll see where we are with time, but i'd like to make something like a cluster_manager or something that is passed into the trainer as well and then linked to a backend.

**Pull request merged**